Tasks/DeployVisualStudioTestAgent/CheckTestAgentInstallation.ps1

```diff
@@ -1,14 +1,47 @@
-function Check-Installation($ProductVersion)
+function Check-Installation($ProductVersion, $VersionToInstall)
 {
     $InstalledCheckRegKey = ("SOFTWARE\Microsoft\DevDiv\vstf\Servicing\{0}\testagentcore" -f $ProductVersion)
 	$InstalledCheckRegValueName = "Install"
 	$InstalledCheckRegValueData = "1"
 
-	$isProductExists = Get-ProductEntry -InstalledCheckRegKey $InstalledCheckRegKey -InstalledCheckRegValueName $InstalledCheckRegValueName -InstalledCheckRegValueData $InstalledCheckRegValueData
+	$isProductExists = $false
 
+	$installValue = Get-ProductEntry -InstalledCheckRegKey $InstalledCheckRegKey -InstalledCheckRegValueName $InstalledCheckRegValueName
+
+	if($installValue -and ($installValue -eq $InstalledCheckRegValueData))
+	{
+		$isProductExists = $true
+	}
+	
 	if($isProductExists)
 	{
-		Write-Verbose -Message ("Test Agent already exists") -verbose
+		$versionInstalled = (Get-ProductEntry -InstalledCheckRegKey $InstalledCheckRegKey -InstalledCheckRegValueName "Version")
+		if(!([System.String]::IsNullOrEmpty($VersionToInstall)))
+		{
+			$VersionToInstall = $VersionToInstall.SubString(0, $VersionToInstall.LastIndexOf('.'))
+
+			Write-Verbose -Message ("Version already installed: " + $VersionInstalled) -verbose
+			Write-Verbose -Message ("Version to be installed: " + $VersionToInstall) -verbose
+		}
+	}
+
+	if($isProductExists)
+	{
+		if(!([System.String]::IsNullOrEmpty($VersionToInstall)))
+		{
+			if(($versionInstalled -ne $null) -and ([version]$VersionToInstall -le [version]$versionInstalled))
+			{
+				Write-Verbose -Message ("Test Agent already exists") -verbose
```

Comment:
>Write-Verbose -Message ("Test Agent already exists") -verbo [](start = 4, length = 59)

can we be explicit on the condition here ? (that we are trying to install something older than existing one ?